gv.c

```diff
@@ -797,10 +799,30 @@ S_gv_fetchmeth_internal(pTHX_ HV* stash, SV* meth, const char* name, STRLEN len,
         cstash = gv_stashsv(linear_sv, 0);
 
         if (!cstash) {
-	    Perl_ck_warner(aTHX_ packWARN(WARN_SYNTAX),
-                           "Can't locate package %" SVf " for @%" HEKf "::ISA",
-			   SVfARG(linear_sv),
-                           HEKfARG(HvNAME_HEK(stash)));
+            if ( ckWARN(WARN_SYNTAX)) {
+                if(     /* these are loaded from Perl_Gv_AMupdate() one way or another */
+                           ( len    && name[0] == '(' )  /* overload.pm related, in particular "()" */
+                        || ( len==7 && strEQ( name, "DESTROY") )
+                ) {
+                     Perl_ck_warner(aTHX_ packWARN(WARN_SYNTAX),
+                            "Can't locate package %" SVf " for @%" HEKf "::ISA",
+                            SVfARG(linear_sv),
+                            HEKfARG(HvNAME_HEK(stash)));
+
+                } else if( len==8 && strEQ( name, "AUTOLOAD") ) {
```

Comment:
Here and elsewhere, use strEQs() when one of the operands is a literal string.  You then don't have to count the length, and it is more maintainable later.

Reply:
I imagine the original author (Yves) did this because he already has the length and it's faster to compare the length and then doing the `strEQ`. On the other hand, we already are in an error case, so micro-optimizing should maybe take a backseat in favour of legibility...

Reply:
strEQs may not have existed then.  It expands to do the same thing as this, so there's no slowdown, and also you're right about not worrying about doing things the most efficient way possible in an error leg.  But again, strEQs doesn't slow things down.  It is more legible and removes the burden of counting from you and future maintainers

Reply:
I also view it as a micro optimization, I wonder if we could add the benefit of it in an optimized macro `strEQsize`

Reply:
Actually, I'm wrong.  There is no strEQs.  There is a memEQs, and you would use it like so

memEQs(name, len, "AUTOLOAD")

Reply:
and it generates the same code, so no slowdown, but improved legibility and maintainability

Reply:
Thanks for the reviews again! I've force-pushed a new version that uses `memEQs`, as the `len` value is already available.